app/views/signup.py

```diff
@@ -64,30 +64,37 @@ def post(self, request, signup_invite: SignupInvite):
 class GameSignupView(View):
     def get(self, request):
         game = most_recent_game()
-        forced_role = SignupInvite.objects.filter(used_at__isnull=False, email=request.user.email).get().player_role
+        forced_role = SignupInvite.objects.filter(used_at__isnull=False,
+                                                  email=request.user.email).get().participant_role
 
-        if not game.is_finished and not request.user.player_set.filter(game=game, active=True).exists():
-            return render(request, "registration/game_signup.html", {'game': game, 'player_role': forced_role})
+        if not game.is_finished and not request.user.participant_set.filter(game=game, active=True).exists():
+            return render(request, "registration/game_signup.html", {'game': game, 'participant_role': forced_role})
         else:
             messages.warning(request, "You're already signed up for the game.")
             return redirect('dashboard')
 
     def post(self, request):
         game = most_recent_game()
-        forced_role = SignupInvite.objects.filter(used_at__isnull=False, email=request.user.email).get().player_role
+        forced_role = SignupInvite.objects.filter(used_at__isnull=False,
```

Comment:
add a check if exists here to avoid server errors if someone posts. Maybe copy from UserSignupVIew? I think we have code there.

Add a check for game state as well

Reply:
okay after further investigation, I'd like to pair up and make the changes to `GameSignupView` and related stuff in a separate PR to merge this one faster, as this is a bit out of scope and I only wanted to add this change since it is directly related to the contents of this PR.